newrelic/python_agent.go

```diff
@@ -0,0 +1,71 @@
+package newrelic
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/buildpacks/libcnb"
+	"github.com/paketo-buildpacks/libpak"
+	"github.com/paketo-buildpacks/libpak/bard"
+	"github.com/paketo-buildpacks/libpak/effect"
+	"github.com/paketo-buildpacks/libpak/sherpa"
+)
+
+type PythonAgent struct {
+	BuildpackPath    string
+	ApplicationPath  string
+	Executor         effect.Executor
+	LayerContributor libpak.DependencyLayerContributor
+	Logger           bard.Logger
+}
+
+func NewPythonAgent(applicationPath string, buildpackPath string, dependency libpak.BuildpackDependency, cache libpak.DependencyCache) (PythonAgent, libcnb.BOMEntry) {
+	contributor, entry := libpak.NewDependencyLayer(dependency, cache, libcnb.LayerTypes{
+		Launch: true,
+	})
+	return PythonAgent{
+		ApplicationPath:  applicationPath,
+		BuildpackPath:    buildpackPath,
+		Executor:         effect.NewExecutor(),
+		LayerContributor: contributor,
+	}, entry
+}
+
+func (p PythonAgent) Contribute(layer libcnb.Layer) (libcnb.Layer, error) {
+	p.LayerContributor.Logger = p.Logger
+
+	p.Logger.Bodyf("Installing to %s", layer.Path)
+
+	fmt.Println("Installing New Relic Python Agent using pip3")
+	if err := p.Executor.Execute(effect.Execution{
+		Command: "pip3",
+		Args:    []string{"install", "newrelic==" + os.Getenv("NEWRELIC_PYTHON_VERSION")},
+		Dir:     p.ApplicationPath,
+		Stdout:  p.Logger.InfoWriter(),
+		Stderr:  p.Logger.InfoWriter(),
```

Comment:
```suggestion
		Stdout:  bard.NewWriter(p.Logger.InfoWriter(), bard.WithIndent(3)),
		Stderr:  bard.NewWriter(p.Logger.InfoWriter(), bard.WithIndent(3)),
```

The API for this is not great, but I like to set these up to indent a little. It'll basically push all the output from the command run over so that it lines up with the other output from the buildpack.

Reply:
I have decided to allow the users to install the New Relic agent using paketo/python-pip buildpack by configuring their requirements.txt in route of their app. That said it makes this configuration redundant.